packages/paste-core/components/toggle/src/index.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import type {ToggleProps} from './types';
+import type {BackgroundColor, TextColor} from '@twilio-paste/style-props';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {useUID} from '@twilio-paste/uid-library';
+import {MediaObject, MediaFigure, MediaBody} from '@twilio-paste/media-object';
+import {RequiredDot} from '@twilio-paste/label';
+import {ToggleKnob} from './ToggleKnob';
+import {ToggleHelpText} from './ToggleHelpText';
+import {ToggleLabelText} from './ToggleLabelText';
+
+const Toggle = React.forwardRef<HTMLDivElement, ToggleProps>(
+  (
+    {element = 'TOGGLE', children, helpText, id, disabled = false, on = false, required = false, onClick, ...props},
+    ref
+  ) => {
+    const [toggleIsOn, setToggleIsOn] = React.useState(on);
+    const [isHovering, setIsHovering] = React.useState(false);
+
+    if (!id) {
```

Comment:
suggestion: just as a general rule of thumb to prevent any weird issues in JS - because function arguments and props [can be passed either by reference or by value](https://stackoverflow.com/questions/518000/is-javascript-a-pass-by-reference-or-pass-by-value-language) - don't make changes to variables defined from the function signature.  So for here I'd suggest doing 
```js
const toggleId = id ? id : useUID();
```
without modifying `id` directly